okhttp/src/main/java/okhttp3/internal/http/RealInterceptorChain.java

```diff
@@ -92,14 +92,19 @@ public RealInterceptorChain(List<Interceptor> interceptors, Transmitter transmit
 
   @Override public Interceptor.Chain withWriteTimeout(int timeout, TimeUnit unit) {
     int millis = checkDuration("timeout", timeout, unit);
-    return new RealInterceptorChain(interceptors, transmitter, connection, index, request, call,
+    return new RealInterceptorChain(interceptors, transmitter, exchange, index, request, call,
         connectTimeout, readTimeout, millis);
   }
 
   public Transmitter transmitter() {
     return transmitter;
   }
 
+  public Exchange exchange() {
+    if (exchange == null) throw new IllegalStateException();
```

Comment:
that's weird, isn't it?
Field is final and nullable but you don't allow access to it in some cases? How can the caller know it's good to call it?